packages/webdriver-utils/src/providers/genericProvider.js

```diff
@@ -164,9 +169,12 @@ export default class GenericProvider {
 
   async getTag() {
     if (!this.driver) throw new Error('Driver is null, please initialize driver with createDriver().');
-    const { width, height } = await this.metaData.windowSize();
+    let { width, height } = await this.metaData.windowSize();
     const resolution = await this.metaData.screenResolution();
     const orientation = this.metaData.orientation();
+    [this.header, this.footer] = await this.getHeaderFooter();
+    // for android window size only constitutes of browser viewport, hence adding nav / status / url bar heights
+    height = this.capabilities.platformName && this.capabilities.platformName.toLowerCase() === 'android' ? height + this.header + this.footer : height;
```

Comment:
Not all the devices will have this please confirm that this will not break for `ios` and `desktop`. Also do we not need to do this on `ios`?

Reply:
Check `resolver` once where we check divide the flow between `mobileResolver` and `desktopResolver`, check that flow once as well.

Reply:
Using `metaData.osName()` now.
verified that it is working as expected and is present for both desktop and mobile metadata resolvers.